### PR TITLE
Focus object with better camera position and transition animation

### PR DESCRIPTION
**Pull request by vincentfretin**
This closes:
- #797 
- #693 
- #798 
- #799

### CODE REVIEW

File: src/editor/lib/EditorControls.js [hunk 1]
```diff
@@ -76,12 +83,73 @@ THREE.EditorControls = function (_object, domElement) {
...
+          this.transitioning = false;
+          object.position.copy(this.transitionCamPosEnd);
+          object.quaternion.copy(this.transitionCamQuaternionEnd);
+          scope.dispatchEvent(changeEvent);
+        }
+      }
+    }
   };
+  // Register the tick function with the render loop
+  AFRAME.scenes[0].addBehavior(this.fakeComponent);
```
Comment (vincentfretin) [inline src/editor/lib/EditorControls.js:152]: Note that addBehavior is not a public aframe api. The fakeComponent here with the isPlaying flags works on aframe 1.5.0, but it doesn't seem to work on aframe master, the tick is not called. I know there was some changes to behaviors with component order. I'm investigating it. I'll do another PR once I found something that works on both 1.5.0 and master, if this one is already merged.

**Comment by vincentfretin** [comment]
The logic of the transition animation with the slerp was taken from https://github.com/c-frame/aframe-cursor-teleport/blob/master/index.js so I don't expect any issue.

File: src/editor/lib/EditorControls.js (hunk 1 above)
Comment (kfarr) [reply src/editor/lib/EditorControls.js:152]: noted, thanks @vincentfretin for the heads up

Comment (vincentfretin) [reply src/editor/lib/EditorControls.js:152]: I tried to register a real component, then you need to add an entity to the scene with this component, then you need to handle for it to play, and keep playing if you switch back and forth between viewer and editor mode. That's more brain wracking than continuing with this hack, at least the changes are isolated there. I fixed it for aframe 1.6.0 in #817

**Pull request merged**